elytron/WFCORE-4362-configurable-certificate-authority-in-certificate-authority-account.adoc

```diff
@@ -0,0 +1,66 @@
+= Make the certificate authority used by a certificate-authority-account configurable
+:author:            Farah Juma, Diana Vilkolakova
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently, Let's Encrypt (https://acme-v02.api.letsencrypt.org/directory) is the only certificate authority
+that can be used in a `certificate-authority-account` resource.
+This issue aims to make it possible for the user to configure attribute `certificate-authority` for this resource.
+Each certificate authority can be associated with a production url and a staging url, both of which have to be configurable.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4362[WFCORE-4362]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1150
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Vilkolakova]
+
+=== QE Contacts
+
+* mailto:mchoma@redhat.com[Martin Choma]
+
+=== Affected Projects or Components
+
+* Wildfly Core
+* Elytron
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+It must be possible to specify staging url and production url for the attribute `certificate-authority` in the `certificate-authority-account` resource.
+
+== Implementation Plan
+
+`Certificate-authority-account` contains optional attribute `certificate-authority` of type STRING.
+The default and only allowed value is "LetsEncrypt".
+We should be able to specify staging URL and production URL without breaking backwards compatibility.
+
+In the Elytron subsystem, LetsEncrypt is a value of CertificateAuthority enum. This enum contains name, url and staging url.
+We can add new `certificate-authority` resource to the elytron
+subsystem with non-optional attributes `name`, `production-url` and optional attribute `staging-url`.
+This way, during the configuration of `certificate-authority-account`
+it will be possible to reference any `certificate-authority` previously added by attribute `name`.
+If no attribute `certificate-authority` is referenced when configuring `certificate-authority-account`, LetsEncrypt can still be used as default.
```

Comment:
I think it would make sense for the default configuration to have the following specified for Let's Encrypt:

```
<certificate-authorities>
    <certificate-authority name="LetsEncrypt" production-url="https://acme-v02.api.letsencrypt.org/" staging-url="https://acme-staging-v02.api.letsencrypt.org/" />
</certificate-authorities>
```

The ```certificate-authority``` attribute for the ```certificate-authority-account``` resource could then continue to have "LetsEncrypt" as a default value.